<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

In [2]:
while FALSE:
    print('*', end='')

NameError: name 'FALSE' is not defined

In [1]:
# print = 'PRINT'
print('Hello World')
print('WHYYYYYYYYYY')

Hello World
WHYYYYYYYYYY


In [ ]:
squares = [i**2 for i in range(10)]
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [7]:
%%timeit
squares = [i**2 for i in range(10)]

895 ns ± 50.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [ ]:
squares = list()
for i in range(10):
    squares.append(i**2)
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [8]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

1.24 μs ± 70.6 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


List of squares are the same
1st one is 895 nanoseconds per loop, 2nd is 1.24 microseconds per loop. Time taken for interpreter to execute each block of code, 2nd block is less efficient than the first as it involves more operations.

## Footnotes